File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/infrastruktur/MoneyConverter.java
package org.github.haschi.haushaltsbuch.infrastruktur;

import cucumber.api.Transformer;
import org.github.haschi.haushaltsbuch.api.Währungsbetrag;

public class MoneyConverter extends Transformer<Währungsbetrag>
{
    @Override
    public final Währungsbetrag transform(final String währungsbetrag)
    {
        return Währungsbetrag.parse(währungsbetrag);
    }
}

File: haushaltsbuch-domain/src/test/kotlin/org/github/haschi/infrastruktur/MoneyConverter.kt
package org.github.haschi.infrastruktur

import cucumber.api.Transformer
import org.github.haschi.domain.haushaltsbuch.modell.core.values.Währungsbetrag

class MoneyConverter : Transformer<Währungsbetrag>()
{

    override fun transform(währungsbetrag: String): Währungsbetrag
    {
        return Währungsbetrag.währungsbetrag(währungsbetrag)
    }
}
